src/event_group_rule.py

```diff
@@ -0,0 +1,18 @@
+""" Module for defining a rule to group together events """
+
+class EventGroupRule:
+    """ Defines the tags that make up an event group and any conextual
+        events that should be parsed as well. Also defines if
+        statistics should be collected for this specific event group
+        
+        Attributes
+        tag: label for the rule
+        trigger_event_tags: list of tags associated to LogEvents
+                            and that define the event group
+        context_event_tags: list of tags associated to LogEvents
+                            and that supply context to event group """
+
+    def __init__(self, tag, trigger_event_tags, context_event_tags=[]):
```

Comment:
Using a mutable default argument like this is tricky. If you construct an EventGroupRule and it uses the default argument for context_event_tags then self.context_event_tags will be assigned a _reference_ to the default value. Which means that modifying self.context_event_tags later would change the default value for future uses.

The safer option is generally either to make a copy of context_event_tags when saving it off, or to default context_event_tags to None and then do something like a "if context_event_tags is None then [] else context_event_tags" when setting self.context_event_tags.

Reply:
I didn't know this! I'll change that in a couple other places as well.